### PR TITLE
Fixed cql_filter parsing for double quotes

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Support for parsing cql 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7491

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/epics/wfsquery.js [hunk 1]
```diff
@@ -178,9 +178,8 @@ export const wfsQueryEpic = (action$, store) =>
             // getSelected Layer and merge layerFilter and cql_filter in params  with action filter
             const layer = selectedLayerSelector(store.getState());
 
-            const {layerFilter, params} = layer ?? {};
-            const cqlFilter = find(Object.keys(params || {}), (k = "") => k.toLowerCase() === "cql_filter");
-
+            const {layerFilter, params = {}} = layer ?? {};
+            const cqlFilter = params?.[find(Object.keys(params || {}), (k = "") => k.toLowerCase() === "cql_filter")];
```
Comment (MV88) [inline web/client/epics/wfsquery.js:182]: why the need of k = "" , do we have keys null or undefined?

Comment (MV88) [reply web/client/epics/wfsquery.js:182]: also can you add unit tests here, i'm 100% convinced about this refactor by only looking at the code

**Review by MV88 — CHANGES_REQUESTED** [review]
@offtherailz 
a few things when you can see them

File: web/client/utils/ogc/Filter/CQL/parser.js [hunk 2]
```diff
@@ -16,7 +16,7 @@ const spatialOperators = {
     WITHIN: "WITHIN"
 };
 const patterns = {
-    PROPERTY: /^[_a-zA-Z]\w*/,
+    PROPERTY: /^"?[_a-zA-Z"]\w*"?/,
```
[the remark this answers is not in the document]
Comment (offtherailz) [reply web/client/utils/ogc/Filter/CQL/parser.js:19]: No. double quotes is not reserved char for regex.
Source: https://www.regular-expressions.info/characters.html

File: web/client/epics/wfsquery.js (hunk 1 above)
Comment (offtherailz) [reply web/client/epics/wfsquery.js:182]: Not a refactor but a fix. In fact the original was not working (taking the key "cql_filter", not the value). 

Need to add a test for this.

Comment (offtherailz) [reply web/client/epics/wfsquery.js:182]: added a unit test :+1:

**Review by MV88 — APPROVED** [review]

**Comment by MV88** [comment]
@ElenaGallo please test in DEV

**Pull request merged**